libs/features/mentor-courses/src/lib/features-mentor-courses.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react';
+import { SharedCourseCard } from '@ppm/shared/course-card';
+import { useSelector, useDispatch } from 'react-redux';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { createStructuredSelector } from 'reselect';
+import { coursesActions, coursesSelectors } from '@ppm/data-access/courses';
+import { likesActions } from '@ppm/data-access/likes';
+import {
+  userProfileActions,
+  userProfileSelectors,
+} from '@ppm/data-access/user-profile';
+import { SharedCourseDetails } from '@ppm/shared/course-details';
+import { SharedCourseInfo } from '@ppm/shared/course-info';
+import { mentorsActions, mentorsSelectors } from '@ppm/data-access/mentors';
+// import { likesActions, likesSelectors } from '@ppm/data-access/likes';
+import './features-mentor-courses.scss';
+import { LikeEnum } from 'libs/data-access/likes/src/lib/types';
+import { RouteComponentProps, RouteProps } from 'react-router-dom';
+
+interface RouteInfo extends RouteProps {
+  params: {
+    id: string;
+  };
+  path?: string | string[];
+}
+
+const stateSelector = createStructuredSelector({
+  courses: coursesSelectors.selectCourses(),
+  loading: coursesSelectors.selectLoading(),
+  profile: userProfileSelectors.selectUserProfile(),
+  loadedProfile: userProfileSelectors.selectOtherProfile(),
+});
+
+export const FeaturesMentorCourses = (props: {
+  history: History;
+  match: RouteComponentProps<RouteInfo>;
+}) => {
+  const dispatch = useDispatch();
+  const { courses, loading } = useSelector(stateSelector);
+  const [coursesState, setCoursesState] = useState([]);
+  const { loadedProfile } = useSelector(stateSelector);
+  const [categories, setCategories] = useState([]);
+  const { profile } = useSelector(stateSelector);
+  const { mentors } = useSelector(stateSelector);
+  const saveClick = (payload: any) => {
+    const data = { callback: 'getAll', ...payload };
+    dispatch(coursesActions.smallUpdate(data));
+  };
+
+  const likeClick = (courseId: string, type: string) => {
+    dispatch(likesActions.create({ course: courseId, type: type }));
+    const clickedCourse = coursesState.find((item) => item._id === courseId);
+    const foundItem = clickedCourse.likesList.findIndex(
```

Comment:
if you needed only boolean statement you can use **some** function instead **findIndex**
and then below change **foundItem !== -1** to **foundItem**
findIndex using only when you needed find index in array

Reply:
Honestly, according to new implementation this will not be necessary at all